pkg/optee-os/Dockerfile

```diff
@@ -0,0 +1,132 @@
+
+# hadolint ignore=DL3006
+FROM target-${TARGETARCH}-build-${BUILDARCH} AS build
+
+SHELL ["/bin/bash", "-eo", "pipefail", "-c"]
+
+RUN mkdir /optee_os /opteeos_bins
+
+# Patches
+COPY patches /tmp/patches
+
+# RSA keys to sign TAs
+COPY eve-optee.pem.gz  /optee_os
+COPY eve-optee-pub.pem.gz /optee_os
+
+WORKDIR /optee_os
+
+RUN gunzip eve-optee.pem.gz && gunzip eve-optee-pub.pem.gz
+ENV EVE_TA_SIGN_KEY="/optee_os/eve-optee.pem"
+
+# Target platforms to build OPTEE-OS
+ENV OPTEEOS_TARGETS="qemu imx8mp_pollux imx8mp_epc_r3720"
+
+# QEMU
+ENV OPTEEOS_REV_qemu=2be3770e85826ae69748fc3e68920d7293a98c5d
+ENV OPTEEOS_qemu="vexpress-qemu_armv8a"
+ENV OPTEEOS_FLAVOR_qemu=""
+ENV OPTEEOS_UART_qemu="0x7ff70000"
+ENV OPTEEOS_SRC_qemu="qemu"
+ENV OPTEEOS_CUSTOM_qemu=""
+
+# i.MX8MP
+ENV OPTEEOS_REV_imx8mp_pollux=c939619d64dea014ad1b8382356eee4d1cbfbb22
+ENV OPTEEOS_imx8mp_pollux="imx"
+ENV OPTEEOS_FLAVOR_imx8mp_pollux="mx8mpevk"
+ENV OPTEEOS_UART_imx8mp_pollux="0x30860000"
+ENV OPTEEOS_SRC_imx8mp_pollux="pollux"
+ENV OPTEEOS_CUSTOM_imx8mp_pollux=""
+
+ENV OPTEEOS_REV_imx8mp_epc_r3720=c939619d64dea014ad1b8382356eee4d1cbfbb22
+ENV OPTEEOS_imx8mp_epc_r3720="imx"
+ENV OPTEEOS_FLAVOR_imx8mp_epc_r3720="mx8mpevk"
+ENV OPTEEOS_UART_imx8mp_epc_r3720="0x30880000"
+ENV OPTEEOS_SRC_imx8mp_epc_r3720="epc_r3720"
+ENV OPTEEOS_CUSTOM_imx8mp_epc_r3720="CFG_DDR_SIZE=0x180000000ULL"
+
+# Repositories
+ADD https://github.com/OP-TEE/optee_os.git#${OPTEEOS_REV_qemu} ${OPTEEOS_SRC_qemu}
+ADD https://github.com/nxp-imx/imx-optee-os.git#${OPTEEOS_REV_imx8mp_epc_r3720} ${OPTEEOS_SRC_imx8mp_epc_r3720}
+ADD https://github.com/nxp-imx/imx-optee-os.git#${OPTEEOS_REV_imx8mp_pollux} ${OPTEEOS_SRC_imx8mp_pollux}
+
+# hadolint ignore=DL3003,SC2086
+RUN for t in ${OPTEEOS_TARGETS}; do \
+        [ "$EVE_TARGET_ARCH" != "aarch64" ] && break ;\
+        target=$(eval echo \$OPTEEOS_${t}) ;\
+        repo=$(eval echo \$OPTEEOS_SRC_${t}) ;\
+        uartbase=$(eval echo \$OPTEEOS_UART_${t}) ;\
+        flavor=$(eval echo \$OPTEEOS_FLAVOR_${t}) ;\
+        customdef=$(eval echo \$OPTEEOS_CUSTOM_${t}) ;\
+        patchfile=$(eval echo /tmp/patches/series-${t}) ;\
+        (cd ${repo} ;\
+         cp ${EVE_TA_SIGN_KEY} keys/default_ta.pem && \
+         [ -f "$patchfile" ] && for p in `cat $patchfile`; \
+             do patch -p1 < "/tmp/patches/${p}"; done ;\
+         make clean && \
+         make -j "$(getconf _NPROCESSORS_ONLN)" \
+            CROSS_COMPILE="${CROSS_COMPILE_ARM32_ENV}" \
+            CROSS_COMPILE64="${CROSS_COMPILE_ENV}" \
+            CFG_TEE_BENCHMARK=n \
+            CFG_TEE_CORE_LOG_LEVEL=1 \
+            PLATFORM="${target}" \
+            PLATFORM_FLAVOR="${flavor}" \
+            CFG_UART_BASE="${uartbase}" \
+            O=out/arm \
+            ${customdef} ;\
+         mkdir -p "/opteeos_bins/${t}" ;\
+         cp out/arm/core/tee-raw.bin "/opteeos_bins/${t}/optee.bin" ;\
+         cp -r out/arm/export-ta_arm64 "/opteeos_bins/${t}/export-ta_arm64") ;\
+    done
```

Comment:
Isn't it possible to split it into several `RUN`s?

Reply:
@christoph-zededa there used to be q yetus complaints when there was a sequence of RUN statements, and I don't know if we turned off that complaint.

Reply:
We turned it off: https://github.com/lf-edge/eve/blob/master/.hadolint.yaml